src/algebra/algebra/ordered.lean

```diff
@@ -0,0 +1,76 @@
+/-
+Copyright (c) 2020 Scott Morrison. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Scott Morrison
+-/
+
+import algebra.algebra.basic
+import algebra.module.ordered
+
+/-!
+# Ordered algebras
+
+An ordered algebra is an ordered semiring, which is an algebra over an ordered commutative semiring,
+for which scalar multiplication is "compatible" with the two orders.
+
+The prototypical example is 2x2 matrices over the reals or complexes (or indeed any C^* algebra)
+where the ordering the one determined by the positive cone of positive operators,
+i.e. `A ≤ B` iff `B - A = star R * R` for some `R`.
+
+## Tags
+
+ordered algebra
+-/
+
+
+/--
+An ordered algebra over an ordered commutative semiring is an algebra
+for which the scalar multiplication is compatible with the orders.
+-/
+@[protect_proj, ancestor semimodule]
+class ordered_algebra (R A : Type*)
+  [ordered_comm_semiring R] [ordered_semiring A] extends algebra R A :=
+(smul_lt_smul_of_pos : ∀ {a b : A}, ∀ {r : R}, a < b → 0 < r → r • a < r • b)
+(lt_of_smul_lt_smul_of_nonneg : ∀ {a b : A}, ∀ {r : R}, r • a < r • b → 0 ≤ r → a < b)
```

Comment:
I'm not suggesting you change it, but if this were 0 < r (which is equivalent) then these two axioms become 0 < r -> (a < b <-> ra < rb).

Reply:
From my recent `ordered_semimodule` patch: it suffices to ask for `r • a < r • b → 0 ≤ r → a ≤ b`. While this is equivalent, sometimes it's easier to show this implication. And I suggest that you turn `ordered_semimodule` into a `Prop` mixin, and use it instead of `ordered_algebra` (extra axioms are **exactly the same**).

Reply:
Thanks @urkud, this is a great suggestion. I have made `ordered_semimodule` a mixin in #4719, and will use that instead.